File: POO/po01a/class/voiture.php

<?php
class voiture
{
    var $marque;
    var $modele;
    var $nom;
    var $compteur = 0;
    var $demarre = false;

    function demarrer()
    {
        $this->demarre = true;
        echo "<p>la voiture a demarré !</p>";
    }
    function avancer($km)
    {
        if ($this->demarre == true) {
            $this->compteur = $this->compteur + $km;
            echo "<p>la voiture a parcouru " . $km . " km </p>";
        } else {
            echo "<p>il faut demarrer la voiture avant !</p>";
        }
    }
    function arreter()
    {
        $this->demarre = false;
        echo "<p>la voiture c'est arréter </p>";
    }
    function afficher()
    {
        if ($this->demarre) {
            $lib = "oui";
        } else {
            $lib = "non";
        }
        echo "<ul>";
        echo "<li>Marque : " . $this->marque . "</li>";
        echo "<li>Modele : " . $this->modele . "</li>";
        echo "<li>Nom : " . $this->nom . "</li>";
        echo "<li>compteur : " . $this->compteur . "</li>";
        echo "<li>A demarrer : " . $lib . "</li>";
        echo "</ul>";
    }
}

File: JAVA/Java40/src/java40/Voiture.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package java40;

/**
 *
 * @author MA500484
 */
public class Voiture {

    String marque;
    String modele;
    String nom;
    int compteur = 0;
    boolean demarre = false;
        
    public void demarrer() {
        demarre = true;

        System.out.println("la voiture a demarré");
    }

    public void avancer(int km) {
        if (demarre == true) {
            compteur = compteur + km;
            System.out.println("la voiture a parcouru" + km);
        } else {
            System.out.println("il faut demarrer la voiture avant !");
        }
    }

    public void arreter() {
        demarre = false;

        System.out.println("la voiture c'est arréte");
    }

    public void afficher() {
        if (demarre) {
            String lib = "oui";
        } else {
            String lib = "non";
        }
        System.out.println("-- information de la voiture --");
        System.out.println("marque: "+ marque);
        System.out.println("modele: "+ modele);
        System.out.println("compteur"+compteur);
        System.out.println("moteur demarrer ? "+demarre);
        
    }
}
